extensions/wikia/CreateNewWiki/js/CommunityBuilderOptIn.js

```diff
@@ -0,0 +1,130 @@
+define('ext.createNewWiki.communityBuilderOptIn', [], function() {
+	'use strict';
+
+	var currentLanguage,
+		optInPrompts,
+		sectionWrapper,
+		optInModal,
+		nameInput,
+		domainInput,
+		viewedOptInModal = false,
+		PROMPT_ENABLED_WRAPPER_CLASS = 'with-optin',
+		ENABLED_ON_LANGUAGES = ['en'],
+		SHOW_MODAL_ON_HUBS = ['1'],
+		SHOW_MODAL_ON_CATEGORIES = ['anime', 'tv'];
+
+	function init($sectionWrapper, $languageInput, $hubInput, $nameInput, $domainInput) {
+		if (!window.AllowCommunityBuilderOptIn) {
+			return;
+		}
+
+		// move the modal because otherwise it can't stack on top of the global nav
+		$(document.body).append($('.optin-modal').detach());
+
+		sectionWrapper = $sectionWrapper;
+		nameInput = $nameInput;
+		domainInput = $domainInput;
+		optInPrompts = $('.optin-prompt');
+		optInModal = $('.optin-modal');
+
+		updateLanguage($languageInput.val());
+
+		$languageInput.on('change', onLanguageInputChange);
+		$hubInput.on('change', onHubInputChange);
+		bindCategoryCheckboxes();
+	}
+
+	function bindCategoryCheckboxes() {
+		var selector = SHOW_MODAL_ON_CATEGORIES
+			.map(function(c) {
+				return 'input[data-short='+c+']';
+			})
+			.join(',');
+
+		$(selector).on('change', onShowModalCategoryChanged);
+	}
+
+	function onLanguageInputChange(event) {
+		updateLanguage(event.currentTarget.value);
+	}
+
+	function onHubInputChange(event) {
+		if (SHOW_MODAL_ON_HUBS.indexOf(event.currentTarget.value) !== -1) {
+			showModal();
+		}
+	}
+
+	function onShowModalCategoryChanged(event) {
+		if (event.currentTarget.checked) {
+			showModal();
+		}
+	}
+
+	function onClickUseCommunityBuilder(event) {
+		event.preventDefault();
+
+		var domainName = domainInput.val().trim(),
+			displayName = nameInput.val().trim(),
+			submit = true;
+
+		if (!domainName || !displayName) {
+			submit = false;
+		}
+
+		var params = [
+			'domain='+encodeURIComponent(domainName),
+			'display='+encodeURIComponent(displayName),
+			'template=TV'
+		];
+
+		if (submit) {
+			params.push('submit=true');
```

Comment:
this seems like CSRF

Reply:
Not sure why this is an issue. Can you explain more?

Reply:
It seems to me that this page will immediately create the new community on a GET request, I think we should be restricting the request method & check some token / or validate Origin

Reply:
It doesn't create the community, it pre-fills the form and tries to submit it, which will fail if the user is anonymous or the input doesn't validate. This effectively proxies an existing API call `put /.../.../...` with a json body. I don't see an issue with this.

Reply:
thanks for clarifying, so LGTM :)